Documentation/contributing/release/organization.rst

```diff
@@ -31,5 +23,69 @@ for the next stable release. New patch releases for an existing stable version
 New patch releases for stable branches are made periodically to provide
 security and bug fixes, based upon community demand and bugfix severity.
 Potential fixes for an upcoming release are first merged into the ``main``
-branch, then backported to the relevant stable branches using the following
-criteria.
+branch, then backported to the relevant stable branches according to the
+:ref:`backport_criteria`.
+
+The following sections describe in more detail the general guidelines that the
+release management team follows for Cilium. The team may diverge from this
+process at their discretion.
+
+Feature Releases
+~~~~~~~~~~~~~~~~
+
+There are several key dates during the feature development cycle of Cilium
+which are important for developers:
+
+* Prerelease days: The Cilium release management team aims to publish a
+  snapshot of the latest changes in the ``main`` branch on the first weekday of
+  each month. This provides developers a target delivery date to incrementally
+  ship functionality, and allows community members to get early access to
+  upcoming features to test and provide feedback. Prereleases may not be
+  published when a release candidate or final stable release is being
+  published.
+
+* Feature freeze: Around six weeks prior to a target feature release, the
+  ``main`` branch is frozen for new feature contributions. The goal of the
+  freeze is to focus community attention on stabilizing and hardening the
+  upcoming release by prioritizing bugfixes, documentation improvements, and
+  tests. In general, all new functionality that the community intends to
```

Comment:
```suggestion
  tests. In general, all new functionalities that the community intends to
```

Reply:
Interesting, `functionalities` sounds awkward to me. [Stackexchange discussion notes on the topic](https://english.stackexchange.com/questions/196590/functionality-vs-functionalities-are-both-correct-and-idiomatic) seem a bit mixed but generally come to the conclusion that either approach may be idiomatic in common English. :thinking: